downloader/src/main/kotlin/Downloader.kt

```diff
@@ -302,7 +302,12 @@ object Downloader {
                     // So first look for a dedicated header in the response, but then also try both redirected and
                     // original URLs to find a name which has a recognized archive type extension.
                     response.headers("Content-disposition").mapNotNullTo(candidateNames) { value ->
-                        value.withoutPrefix("attachment; filename=")
+                        val tokens: List<String> = value.split(";")
+                        when {
+                            // skip first token corresponding to disposition-type (see rfc6266)
+                            tokens.size > 1 -> tokens[1].trim().withoutPrefix("filename=")
+                            else -> tokens[0]
+                        }
```

Comment:
How about this instead?

```
val filenames = value.split(';').mapNotNull { it.trim().withoutPrefix("filename=") }
filenames.firstOrNull()?.removeSurrounding("\"")
```

Note that I also added unquoting of filenames.

Reply:
Yes, it's more concise and elegant !